CONTRIBUTING.md

```diff
@@ -61,16 +239,103 @@ Pull requests should not be squash merged in order to keep complete commit messa
 This means that, when addressing review comments in order to keep the pull request in an always mergeable status, you will sometimes need to rewrite your branch's history and then force-push it with `git push --force-with-lease`.
 Useful git commands that can help a lot with this are `git commit --patch --amend` and `git rebase --interactive`. For more details consult the git man pages or online resources like [git-rebase.io](https://git-rebase.io/) or [The Pro Git Book](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).
 
-## Testing changes
+### Submit the change
+
+- If you created a branch based on your current NixOS, [rebase](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) your branch against current `master`.
+
+- Push your changes to your fork of nixpkgs.
+- Create the pull request
+
+(pushing, creating a pull request, select the right branch (link to above section))
+
+When pull requests are made, our tooling automation bot,
```

Comment:
Can you add considerations on mass ping & the fact that a PR should be open draft if it does not meet the minimal requirements of having a descriptive title and description? Apparently it's not clear.

Reply:
In the first iteration of this PR I'm just going to move existing documentation without writing new one, and I want to only limit new docs to holes that need to be filled, otherwise this would quickly get too big. I think this here in particular should be out of scope for this initial PR.

Reply:
In that case, let's not write more things like the `(optional) ` ? :D

Reply:
I do count that as organization, it shows up in the ToC :)

Reply:
Oh but yes I will have to make sure the titles are accurate, so the `(optional)` should be revised for sure

Reply:
Move + changing the appropriate titles makes it hard to review what changed in general :P.
I'd appreciate if there was a commit for all the title renaming stuff or "what is new" compared to the old contents.

Reply:
Sounds like a good idea, I'll consider that when cleaning up the history

Reply:
I cleaned up the history now. Content was strictly only moved, only adding new titles when really necessary, and then wrapping them with () to indicate they need to be considered. For every commit only contents from a single file was moved.